examples/basic/frameworks/haystack_deep_research_agent/src/nat_haystack_deep_research_agent/register.py

```
@@ -0,0 +1,156 @@
+        Returns:
+            Comprehensive research report.
+        """
+        try:
+            logger.info(f"Processing research query: {input_message}")
+
+            # Create messages
+            messages = [ChatMessage.from_user(input_message)]
+            agent_output = agent.run(messages=messages)
+
+            # Extract response
+            if "messages" in agent_output and agent_output["messages"]:
```

Ruff (0.13.1)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)